lib/ASRouter.jsm

```diff
@@ -468,18 +470,37 @@ class _ASRouter {
   _validPreviewEndpoint(url) {
     try {
       const endpoint = new URL(url);
-      if (!WHITELIST_HOSTS[endpoint.host]) {
+      if (!this.WHITELIST_HOSTS[endpoint.host]) {
         Cu.reportError(`The preview URL host ${endpoint.host} is not in the whitelist.`);
       }
       if (endpoint.protocol !== "https:") {
         Cu.reportError("The URL protocol is not https.");
       }
-      return (endpoint.protocol === "https:" && WHITELIST_HOSTS[endpoint.host]);
+      return (endpoint.protocol === "https:" && this.WHITELIST_HOSTS[endpoint.host]);
     } catch (e) {
       return false;
     }
   }
 
+  _loadSnippetsWhitelistHosts() {
+    let additionalHosts = [];
+    try {
+      additionalHosts = JSON.parse(Services.prefs.getStringPref(SNIPPETS_ENDPOINT_WHITELIST));
+    } catch (e) {}
```

Comment:
it might be useful to surface a parsing error if someone is trying to debug this?